data_handling/synchronized_data_logger.py

```diff
@@ -0,0 +1,371 @@
+#!/usr/bin/env python2
+# -*- coding: utf-8 -*-
+"""
+Created on Tue May 30 17:18:42 2017
+
+@author: Zorn, Ronecker
+"""
+
+import numpy as np
+import rospy
+import os
+import sys
+import h5py
+import datetime
+import cv2
+import message_filters
+
+from torcs_msgs.msg import TORCSSensors
+from sensor_msgs.msg import LaserScan
+from sensor_msgs.msg import Image
+from std_srvs.srv import Trigger
+from torcs_msgs.msg import TORCSCtrl
+from cv_bridge import CvBridge, CvBridgeError
+from std_msgs.msg import String
+
+
+data = {}
+data['image']   = []
+data['time'] = []
+class DataLogger:
+    def __init__(self, logfile_name='log_1.h5', sensors_topic='torcs_ros/sensors_state',image_topic='torcs_ros/pov_image',control_topic='/torcs_ros/ctrl_state', scan_track_topic='torcs_ros/scan_track', log_at_start=True):
+        
+        self.sleep_rate = 0.2 #variable used to limit data to every x seconds
+        self.extension ="ben" #define user prefix for file name
+        self.find_dir() #find current working directory
+        self.evalFactor = 1.5 #factor used for eval

```

Comment:
@B-Zorn please make those parameters readable from a yaml-file. An example how to do this can be found in [here](https://github.com/fmirus/torcs_ros/blob/master/torcs_ros_drive_ctrl/config/torcs_ros_drive_ctrl.yaml)